rescue/src/util.rs

```diff
@@ -0,0 +1,45 @@
+use gcd::Gcd;
+use modinverse::modinverse;
+use num::{BigUint, One};
+use p3_field::PrimeField64;
+use sha3::{
+    digest::{ExtendableOutput, Update, XofReader},
+    Shake256,
+};
+
+/// The binomial function from combinatorics, used to compute the number of rounds needed for soundness.
+pub(crate) fn binomial(n: usize, k: usize) -> BigUint {
+    let mut result = BigUint::one();
+    for i in 0..k {
+        result *= n - i;
+        result /= i + 1;
+    }
+
+    result
+}
+
+/// Generate alpha, the smallest integer relatively prime to p − 1.
+pub(crate) fn get_alpha<F: PrimeField64>() -> u64 {
+    let p = F::ORDER_U64;
+
+    (3..p).find(|&a| a.gcd(p - 1) == 1).unwrap()
+}
+
+/// Given alpha, find its multiplicative inverse in Z/⟨p − 1⟩.
+pub(crate) fn get_inverse<F: PrimeField64>(alpha: u64) -> u64 {
+    let p = F::ORDER_U64 as i64;
+    modinverse(alpha as i64, p - 1)
```

Comment:
I think modinverse can work with unsigned ints, could we skip both conversions to `i64`?

Reply:
Ah I tried that, but I get an `Attempted to subtract with overflow` error within `modinverse`